### ISSUE TITLE
New CF7 forms overwrites existing ones instead of creating a new form

### ISSUE THREAD

**Issue by deviationist** [DekodeInteraktiv/hogan-form#18]
This occurs when one tries to create a new form in Contact Form 7. The new form does not get stored as a new form – it overwrites the existing ones.

**How to trigger the error**
Go to: http://nettsteder.regjeringen.no/wp-admin/admin.php?page=wpcf7-new
Here you should see that the shortcode preview field is already present and has an ID filled in (see attached screenshot). This is not right – this field should only be present when editing an existing form.

If you deactivate the Hogan form-plugin the field disappears and the new form gets created like it should.

![screen shot 2018-08-01 at 11 57 08](https://user-images.githubusercontent.com/7861329/43515126-0cb878d4-9582-11e8-9822-fadb3f268ae6.png)

**Comment by deviationist** [comment]
As shown in the code below, the "wp_reset_postdata" is ran after a custom query. But this does not seem to actually reset the query. This seems to be what causes the problem. Because of this query the $post-variable is already set, and it does not get reset (for some reason), and then CF7-form behaves like you are editing a form instead of creating a new one.

https://github.com/DekodeInteraktiv/hogan-form/blob/2d26438c70f3fb0b0ea05261b48be3b7ea118f58/includes/form-providers/class-contactform7-provider.php#L61-L65

**Comment by deviationist** [comment]
But, if we do like this (sorry for ugly code) everything works as i should. The preview field is not present, and the post gets created as a new form as it should.

https://github.com/DekodeInteraktiv/hogan-form/blob/f8041dccf1805d186dff192f4dabfc4b9bbec4a4/includes/form-providers/class-contactform7-provider.php#L47-L68

**Comment by deviationist** [comment]
I suspect that this is happening since the code executes very early in the WP boot process. I did some digging in WP and "wp_reset_postdata" triggers the "reset_postdata"-method in the global WP_Query-object.

If I remove the if-statement from that method then "wp_reset_postdata" actually resets the postdata. Not sure why `$this->post` is empty since we just ran a query...

File: wp-includes/class-wp-query.php:4058-5062
```
public function reset_postdata() {
	if ( ! empty( $this->post ) ) {
		$GLOBALS['post'] = $this->post;
		$this->setup_postdata( $this->post );
	}
}
